cockroachdb/sqlalchemy/dialect.py

```diff
@@ -142,6 +142,12 @@ def get_indexes(self, conn, table_name, schema=None, **kw):
         res = []
         # Map over uniques because it preserves order.
         for name in uniques:
+            # In order to mimic postgresql behaviour, we hide the primary
+            # index from SQLAlchemy. One of the side effects of exposing it
+            # is when comparing the database state against models, for
+            # example when Alembic autogenerates migrations.
+            if name == "primary":
```

Comment:
The primary key is not always named "primary" (and there are tests for this, so the postgres dialect must sometimes return the PK). Looks like we need more research into what exactly is expected here.

Reply:
👍 didn't know it could be named.

So the result for `get_indexes` will never have the primary key, as indicated by the [query used to get indexes](https://github.com/zzzeek/sqlalchemy/blob/master/lib/sqlalchemy/dialects/postgresql/base.py#L2862). There is a separate query to get the primary key (as opposed to how we re-use `get_indexes`).

Reply:
Another thing to look into is whether we can now remove this override of `get_indexes` and fall back to the base postgres dialect version (which uses various `pg_catalog` tables). When I was starting with this we didn't support enough of `pg_catalog`, but maybe we've added enough now.